src/lib.rs

```diff
@@ -185,14 +186,53 @@ impl SubAllocation {
         self.device_memory
     }
 
+    /// Returns the offset of the allocation on the vk::DeviceMemory.
+    /// When binding the memory to a buffer or image, this offset needs to be supplied as well.
     pub fn offset(&self) -> u64 {
         self.offset
     }
 
-    pub fn mapped_ptr(&self) -> *mut std::ffi::c_void {
+    /// Returns the size of the allocation
+    pub fn size(&self) -> u64 {
+        self.size
+    }
+
+    /// Returns a valid mapped pointer if the memory is host visible, otherwise it will return None.
+    /// The pointer already points to the exact memory region of the suballocation, so no offset needs to be applied.
+    pub fn mapped_ptr(&self) -> Option<std::ptr::NonNull<std::ffi::c_void>> {
         self.mapped_ptr
     }
 
+    /// Returns a valid mapped slice if the memory is host visible, otherwise it will return None.
+    /// The pointer already points to the exact memory region of the suballocation, so no offset needs to be applied.
+    pub fn mapped_slice(&self) -> Option<&[u8]> {
+        if let Some(ptr) = self.mapped_ptr() {
+            unsafe {
+                Some(std::slice::from_raw_parts(
+                    ptr.as_ptr() as *mut u8,
```

Comment:
```suggestion
                    ptr.as_ptr() as *const u8,
```
or
```suggestion
                    ptr.as_ptr() as *const _,
```
Bit odd there's no `.as_ptr()` and `.as_mut_ptr()` :shrug:

Reply:
We decided to use `u64` as it's the actual type that is used by both vulkan and d3d12 for the sizes.